apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -22,26 +38,74 @@ if (!('BridgeServiceMixin' in self)) {
   ]);
 
   var ConversationService = {
+    mobileMessageClient: null,
+
     /**
      * Initializes our service using the Service mixin's initService.
      */
     init() {
       this.initService();
+
+      this.mobileMessageClient = bridge.client(
+        'mozMobileMessageShim',
+        new BroadcastChannel('mobile-message-shim-channel')
+      ).plugin(streamClient);
     },
 
     /**
      * Stream that returns everything needed to display conversations in the
      * inbox view.
-     * @param {ServiceStream.<ConversationSummary>} The stream to use in the
-     * implementation: it will be passed autoamtically by the Bridge library.
+     * @param {ServiceStream.<ConversationSummary>} serviceStream The stream to
+     * use in the implementation: it will be passed automatically by the Bridge
+     * library.
      */
-    getAllConversations(stream) {},
+    getAllConversations(serviceStream) {
+      var getThreadsStream = this.mobileMessageClient.stream('getThreads');
+
+      var draftsPromise = Drafts.request().then(() => {
+        // Return draft list that is sorted by timestamp in inverse order.
+        return Drafts.getAllThreadless().sort(
+          (draftA, draftB) => draftB.timestamp - draftA.timestamp
+        );
+      });
+
+      // If conversation stream is cancelled we should cancel internal threads
+      // stream as well.
+      serviceStream.cancel = (reason) => {
+        debug('getAllConversations stream is cancelled: %s', reason);
+        return getThreadsStream.cancel(reason);
+      };
+
+      // assume that we get the data in an inverse sort order
+      getThreadsStream.listen((thread) => {
+        debug('Retrieved conversation with id %s', thread.id);
+
+        draftsPromise.then((drafts) => {
+          // Flush drafts that are created earlier than current thread.
+          while(drafts.length && drafts[0].timestamp >= thread.timestamp) {
+            serviceStream.write(drafts.shift());
+          }
+
+          thread.draft = Drafts.byThreadId(thread.id);
```

Comment:
That's experimental thing, but I think we generally should return thread drafts as well here, what do you think?